libs/perun/services/src/lib/gui-auth-resolver.service.ts

```diff
@@ -1,19 +1,30 @@
 import { Injectable } from '@angular/core';
 import {
   AuthzResolverService,
+  Group,
+  Member,
   PerunBean,
   PerunPolicy,
   PerunPrincipal,
+  Resource,
   RoleManagementRules,
+  User,
+  Vo,
 } from '@perun-web-apps/perun/openapi';
 import { Role } from '@perun-web-apps/perun/models';
 
+interface Privilege {
+  readAuth: boolean;
+  manageAuth: boolean;
+  modes: string[];
+}
+
+type Entity = Vo & Group & Resource & Member & User;
```

Comment:
I am not sure if there should be ampersands.
```suggestion
type Entity = Vo | Group | Resource | Member | User;
```

Reply:
Again here we need union of types, since there are properties such as `userId` and etc. that are not present in all of the types.